Apps/W1/Shopify/test/Logs/ShpfySkippedRecordLogTest.Codeunit.al

```diff
@@ -0,0 +1,838 @@
+    end;
+
+    [Test]
+    procedure UnitTestLogSalesInvoiceWithCustomerNoUsedInShopifyCustomerTemplates()
+    var
+        SalesInvoiceHeader: Record "Sales Invoice Header";
+        Customer: Record Customer;
+        SkippedRecord: Record "Shpfy Skipped Record";
+        Shop2: Record "Shpfy Shop";
+        ShopifyCustomerTemplate: Record "Shpfy Customer Template";
+        PostedInvoiceExport: Codeunit "Shpfy Posted Invoice Export";
+        PaymentTermsCode: Code[10];
+    begin
+        // [SCENARIO] Log skipped record when sales invoice export is skipped because customer no which is used in shopify customer templates.
+        Initialize();
+
+        // [GIVEN] Customer 
+        CreateRandomCustomer(Customer);
+        // [GIVEN]  Shop 
+        CreateShop(Shop2, '');
+        // [GIVEN] Shopify Customer
+        CreateShopifyCustomer(Customer);
+        // [GIVEN] Payment Terms Code
+        PaymentTermsCode := CreatePaymentTerms(Shop2.Code);
+        // [GIVEN] Shopify Customer Template with customer no.
+        CreateShopifyCustomerTemplate(ShopifyCustomerTemplate, Shop2, Customer);
+        // [GIVEN] Sales Invoice for default customer no.
+        CreateSalesInvoiceHeader(SalesInvoiceHeader, ShopifyCustomerTemplate."Default Customer No.", PaymentTermsCode);
+
+        // [WHEN] Invoke Shopify Posted Invoice Export
+        PostedInvoiceExport.SetShop(Shop2.Code);
+        PostedInvoiceExport.ExportPostedSalesInvoiceToShopify(SalesInvoiceHeader);
+
+        // [THEN] Related record is created in shopify skipped record table.
+        SkippedRecord.SetRange("Record ID", SalesInvoiceHeader.RecordId);
+        LibraryAssert.IsTrue(SkippedRecord.FindLast(), 'Skipped record is not created');
+        LibraryAssert.AreEqual(StrSubstNo('Shopify Customer template exists for customer no. %1 shop %2.', Customer."No.", Shop2.Code), SkippedRecord."Skipped Reason", 'Skipped reason is not as expected');
+    end;
+
+    [Test]
+    procedure UnitTestLogSalesInvoiceWithoutSalesLine()
+    var
+        SalesInvoiceHeader: Record "Sales Invoice Header";
+        SalesInvoiceLine: Record "Sales Invoice Line";
+        Customer: Record Customer;
+        SkippedRecord: Record "Shpfy Skipped Record";
+        PostedInvoiceExport: Codeunit "Shpfy Posted Invoice Export";
+        PaymentTermsCode: Code[10];
+    begin
+        // [SCENARIO] Log skipped record when sales invoice export is skipped because it has no sales lines.
+        Initialize();
+
+        // [GIVEN] Customer
+        Customer := ShpfyInitializeTest.GetDummyCustomer();
+        // [GIVEN] Shopify Customer
+        CreateShopifyCustomer(Customer);
+        // [GIVEN] Payment Terms Code
+        PaymentTermsCode := CreatePaymentTerms(Shop.Code);
+        // [GIVEN] Sales Invoice without sales line.
+        CreateSalesInvoiceHeader(SalesInvoiceHeader, Customer."No.", PaymentTermsCode);
+        // [GIVEN] No existing sales lines.
+        if not SalesInvoiceLine.IsEmpty() then
```

Comment:
Why delete all lines and not just the ones that would be related to the invoice?

Reply:
Its because its done this way:

![image](https://github.com/user-attachments/assets/5bb15c8f-58cb-4e47-bcc7-9d3525aa1200)

Theres no filter for the header.

Reply:
This looks like a bug to me...
It should filter on only the passed header lines. It doesn't make sense to check if there are ANY lines that fit that filter, as there will always be invoiced with "legit" lines.
@GediminasGaubys We should fix this. Maybe also make Andrei aware?